### PR TITLE
[M] CANDLEPIN-913: Updated content endpoints to support improved querying

### PR DESCRIPTION
**Pull request by Ceiu**
- Added the "label", "active", and "custom" filters to the query parameters of the GET /owners/{key}/content endpoint
- Added the "owner", "label", "active", and "custom" filters to the query parameters of the GET /content endpoint
- Removed the now-redundant "omit_globals" query parameter from both GET /owners/{key}/content and GET /content endpoints
- Reworked the GET /contents and GET /owner/{key}/contens endpoints to support filtering on content IDs, content labels, and the "active" and "custom" states
- Added a new query builder object for building dynamic content queries
- Updated the manifest generator test extension to create entities in a namespace rather than global to fix multiple issues with manifest generation

### CODE REVIEW

File: src/main/java/org/candlepin/model/ContentQueryBuilder.java [hunk 1]
```diff
@@ -0,0 +1,666 @@
...
+     * @return
+     *  a reference to this query builder
+     */
+    public ContentQueryBuilder addContentIds(Collection<String> contentIds) {
+        if (contentIds == null) {
+            return this;
+        }
+
+        contentIds.stream()
+            .filter(pid -> pid != null && !pid.isBlank())
```
Comment (joshmalbrecht) [inline src/main/java/org/candlepin/model/ContentQueryBuilder.java:95]: Update from `pid` to `cid` for content ID

File: src/main/java/org/candlepin/model/ContentQueryBuilder.java [hunk 2]
```diff
@@ -0,0 +1,666 @@
...
+     * @return
+     *  a reference to this query builder
+     */
+    public ContentQueryBuilder addContentLabels(Collection<String> contentLabels) {
+        if (contentLabels == null) {
+            return this;
+        }
+
+        contentLabels.stream()
+            .filter(name -> name != null && !name.isBlank())
```
Comment (joshmalbrecht) [inline src/main/java/org/candlepin/model/ContentQueryBuilder.java:134]: Update `name` to `label`

File: src/main/java/org/candlepin/model/ContentQueryBuilder.java [hunk 3]
```diff
@@ -0,0 +1,666 @@
...
+
+    /**
+     * Adds one or more content names to use for filtering query output.
+     *
+     * @param contentLabels
+     *  a collection of content names to use to filter query output
+     *
+     * @return
+     *  a reference to this query builder
+     */
```
Comment (joshmalbrecht) [inline src/main/java/org/candlepin/model/ContentQueryBuilder.java:155]: Update content names to content labels

File: src/main/java/org/candlepin/model/ContentQueryBuilder.java [hunk 4]
```diff
@@ -0,0 +1,666 @@
...
+     *  the list of query chunks to receive the query components for filtering on content names
+     *
+     * @param criteriaChunks
+     *  the list of criteria chunks to receive the predicates for filtering on content names
+     *
+     * @param parameters
+     *  the map of parameters to receive the specific query arguments provided to the builder
+     */
+    private void buildContentNameFilterComponents(List<String> queryChunks, List<String> criteriaChunks,
+        Map<String, Object> queryArgs) {
```
Comment (joshmalbrecht) [inline src/main/java/org/candlepin/model/ContentQueryBuilder.java:420]: Update to content labels instead of content name

**Review by joshmalbrecht — CHANGES_REQUESTED** [review]
Looks good! I just had some comments on some small changes

File: src/main/java/org/candlepin/model/ContentQueryBuilder.java [hunk 5]
```diff
@@ -0,0 +1,666 @@
+/*
+ * Copyright (c) 2009 - 2023 Red Hat, Inc.
```
[the remark this answers is not in the document]
Comment (Ceiu) [reply src/main/java/org/candlepin/model/ContentQueryBuilder.java:2]: Fixed

File: src/main/java/org/candlepin/model/ContentQueryBuilder.java (hunk 1 above)
Comment (Ceiu) [reply src/main/java/org/candlepin/model/ContentQueryBuilder.java:95]: Fixed

File: src/main/java/org/candlepin/model/ContentQueryBuilder.java (hunk 2 above)
Comment (Ceiu) [reply src/main/java/org/candlepin/model/ContentQueryBuilder.java:134]: Fixed

File: src/main/java/org/candlepin/model/ContentQueryBuilder.java (hunk 3 above)
Comment (Ceiu) [reply src/main/java/org/candlepin/model/ContentQueryBuilder.java:155]: Fixed

File: src/main/java/org/candlepin/model/ContentQueryBuilder.java (hunk 4 above)
Comment (Ceiu) [reply src/main/java/org/candlepin/model/ContentQueryBuilder.java:420]: Fixed

**Review by joshmalbrecht — APPROVED** [review]

**Review by bakajstep — APPROVED** [review]

**Pull request merged**